lib/util/LocationStatusManager.js

```diff
@@ -0,0 +1,552 @@
+const { locationStatusCollection } = require('../constants');
+
+const actions = {
+    deleteSchedule: 'deleteScheduledResumeService',
+    pause: 'pauseService',
+    resume: 'resumeService',
+};
+
+const supportedServices = ['lifecycle'];
+
+/**
+ * @class LocationStatusManager
+ *
+ * @classdesc manages the pause/resume status of the locations
+ * on different services.
+ */
+class LocationStatusManager {
+    constructor(mongoServices, mongoClient, zkClient, redis, topics, zkDetails, logger) {
+        this._mongoServices = mongoServices;
+        this._mongoClient = mongoClient;
+        this._zkClient = zkClient;
+        this._redis = redis;
+        this._topics = topics;
+        this._zkDetails = zkDetails;
+        this._logger = logger;
+
+        this._locationStatusColl = null;
+
+        this._locationStatusStore = {};
+        this._scheduledResumeJobs = {
+            crr: {},
+            ingestion: {},
+            lifecycle: {},
+        };
+    }
+
+    /**
+     * Creates mongo collection for
+     * storing the location status
+     * @param {function} cb callback
+     * @return {undefined}
+     */
+    _initCollection(cb) {
+        this._mongoClient.createCollection(locationStatusCollection, err => {
+            // in case the collection already exists, we ignore the error
+            if (err && err.codeName !== 'NamespaceExists') {
+                this._logger.error('Could not create mongo collection', {
+                    method: 'LocationStatusManager._initCollection',
+                    collection: locationStatusCollection,
+                    error: err.message,
+                });
+                return cb(err);
+            }
+            this._locationStatusColl = this._mongoClient.collection(locationStatusCollection);
+            return cb();
+        });
+    }
+
+    /**
+     * list all location status documents from mongodb
+     * @param {function} cb callback
+     * @returns {undefined}}
+     */
+    _listCollectionDocuments(cb) {
+        this._locationStatusColl.find({}, (err, cursor) => {
+            if (err) {
+                this._logger.error('Could not list documents', {
+                    method: 'LocationStatusManager._listCollectionDocuments',
+                    collection: locationStatusCollection,
+                    error: err.message,
+                });
+            }
+            return cursor.toArray(cb);
+        });
+    }
+
+    /**
+     * Add valid mongo locations to local store
+     * @param {Object[]} locations list of locations data from mongo
+     * @param {function} cb callback
+     * @returns {undefined}
+     */
+    _getPreviousLocationStates(locations, cb) {
+        const validLocationNames = Object.keys(validLocations);
+        locations
+            .filter(loc => validLocationNames.includes(loc._id))
+            .forEach(loc => {
+                this._locationStatusStore[loc._id] = new LocationStatus(supportedServices, loc.value);
```

Comment:
The initial reason was cause it's a MongoDB document model, which means that it can get accessed by everyone.

But i guess currently the information is only accessed by backbeat and the logic of pause/resume is also very specific to backbeat. I moved the model to backbeat